backend/api/src/DB/database.go

```diff
@@ -18,178 +15,147 @@ import (
 const passMaxLength int = 50
 const userMaxLength int = 20
 
-// variables
-var db *gorm.DB
-
-var (
-	err        error
-	dbName     string
-	dbUser     string
-	dbPassword string
-)
+// structs
+type dbEnvs struct {
+	Name string
+	User string
+	Pass string
+}
 
 // Functions
 
-func getEnvs() {
-	var ok bool
-
-	dbName, ok = os.LookupEnv("POSTGRES_NAME")
-	if !ok {
-		log.Fatalln("POSTGRES_NAME: env variable not found")
-	}
-
-	dbUser, ok = os.LookupEnv("POSTGRES_USER")
-	if !ok {
-		log.Fatalln("POSTGRES_USER: env variable not found")
-	}
-
-	dbPassword, ok = os.LookupEnv("POSTGRES_PASSWORD")
-	if !ok {
-		log.Fatalln("POSTGRES_PASSWORD: env variable not found")
-	}
-}
-
 // SetupDatabase function
-func SetupDatabase() {
+// Returns the db session or an error.
+func SetupDatabase() (*gorm.DB, error) {
 	// get env variables
-	getEnvs()
+	var envs dbEnvs
+	getDbEnvs(&envs)
 	// connect to DB
-	dsn := fmt.Sprintf("host=localhost user=%s password=%s dbname=%s port=5432 sslmode=disable TimeZone=Europe/Stockholm", dbUser, dbPassword, dbName)
-	db, err = gorm.Open(postgres.Open(dsn), &gorm.Config{})
+	dsn := fmt.Sprintf("host=localhost user=%s password=%s dbname=%s port=5432 sslmode=disable TimeZone=Europe/Stockholm", envs.User, envs.Pass, envs.Name)
+	db, err := gorm.Open(postgres.Open(dsn), &gorm.Config{})
 	if err != nil {
-		log.Fatal(err)
+		return nil, err
 	}
 
 	// migrate database models
-	db.AutoMigrate(&User{})
-	db.AutoMigrate(&Command{})
-
-	// create some commands
-	command := &Command{
-		Keyword:     "test",
-		Description: "some description what this command does",
-		Text:        "the actual output when running the command",
-		Link:        "https://github.com/NoahHakansson",
-	}
-	err := CreateCommand(command)
+	err = db.AutoMigrate(&User{})
 	if err != nil {
-		fmt.Println(err.Error())
-	}
-	fmt.Printf("Command: %#v\n", command)
-
-	command = &Command{
-		Keyword:     "hello",
-		Description: "some description what this command does",
-		Text:        "the actual output when running the command",
-		Link:        "",
+		log.Println(err)
 	}
-	err = CreateCommand(command)
+	err = db.AutoMigrate(&Exam{})
 	if err != nil {
-		fmt.Println(err.Error())
+		log.Println(err)
```

Comment:
Change to throw error instead of printing to the log

Reply:
throws finns inte i GO. de har lite annorlunda error handling. Men bättre än throws IMO. I detta fallet ska det bara printas då det inte är en final version än.

borde lagt till en `WIP:`

Reply:
Aja your the expert 😅